hdr-plus: Drop

OPNA2608 opened the pull request:
ZHF: #199919 (fixing long-time failures by removing the package lol)

###### Description of changes

- Last upstream code change that wasn't an upstreamed fix by a Nixpkgs maintainer in 2020-03-10
- `halide` bumps usually break it
  - was broken by Halide 10.0.0, initially fix upstreamed by us but according to Hydra on 2022-04-17 it broke again on Halide 10.0.0 (and it hasn't been building on Hydra since then)
  - has been reported upstream as broken again on Halide 12.0.0 (2021-05-20)
  - we are now on Halide 14.0.0 with no fix merged or even PR'd
- Originally introduced for `megapixels`, which has dropped support for it due to bad results & complications with its integration
  - With this dependency gone, I have no personal motivation to maintain this package either

For these reasons, I'm dropping the package from Nixpkgs.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

SuperSandro2000 commented on pkgs/top-level/aliases.nix:
```diff
@@ -601,6 +601,7 @@ mapAliases ({
   hardlink = throw "hardlink was merged into util-linux since 2019-06-14."; # Added 2022-08-12
   inherit (harePackages) hare harec; # Added 2022-08-10
   hawkthorne = throw "hawkthorne has been removed because it depended on a broken version of love"; # Added 2022-01-15
+  hdr-plus = throw "hdr-plus has been removed: unmaintained, original reason for inclusion is no more"; # Added 2022-11-08
```
```suggestion
  hdr-plus = throw "hdr-plus has been removed because it unmaintained, often breaks and no longer consumed as a dependency"; # Added 2022-11-08
```

OPNA2608 replied:
Done.

OPNA2608 committed:
hdr-plus: Drop

- Last upstream code change that wasn't an upstreamed fix by a Nixpkgs maintainer in 2020-03-10
- `halide` bumps usually break it
  - was broken by Halide 10.0.0, initially fix upstreamed by us but according to Hydra on 2022-04-17 it broke again on Halide 10.0.0 (and it hasn't been building on Hydra since then)
  - has been reported upstream as broken again on Halide 12.0.0 (2021-05-20)
  - we are now on Halide 14.0.0 with no fix merged or even PR'd
- Originally introduced for `megapixels`, which has dropped support for it due to bad results & complications with its integration
  - With this dependency gone, I have no personal motivation to maintain this package either

For these reasons, I'm dropping the package from Nixpkgs.
